extensions/kafka.md

```diff
@@ -1,91 +1,437 @@
+
+Albeit the default, this implementation allows for some customization,
+ such as how the `EventMessage`'s `MetaData` is mapped to Kafka headers.
+This is achieved by adjusting the "header value mapper" in the `DefaultKafkaMessageConverter`'s builder.
+
+The `SequencingPolicy` can be adjusted to change the behaviour of the record key being used.
+The default sequencing policy is the `SequentialPerAggregatePolicy`,
+ which leads to the aggregate identifier of an event being the key of a `ProducerRecord` and `ConsumerRecord`. 
+
+Lastly, the `Serializer` used by the converter can be adjusted. 
+See the [Serializer](../operations-guide/production-considerations/serializers.md) section for more details on this. 
+
+```java
+public class KafkaMessageConversationConfiguration {
+    // ...
+    public KafkaMessageConverter<String, byte[]> kafkaMessageConverter(Serializer serializer,
+                                                                       SequencingPolicy<? super EventMessage<?>> sequencingPolicy,
+                                                                       BiFunction<String, Object, RecordHeader> headerValueMapper) {
+        return DefaultKafkaMessageConverter.builder()
+                                           .serializer(serializer)                  // Hard requirement
+                                           .sequencingPolicy(sequencingPolicy)      // Defaults to a "SequentialPerAggregatePolicy"
+                                           .headerValueMapper(headerValueMapper)    // Defaults to "HeaderUtils#byteMapper()"
+                                           .build();
+    }
+    // ...
+}
+```
+
+Make sure to use an identical `KafkaMessageConverter` on both the producing and consuming end,
+ as otherwise exception upon deserialization should be expected.
+
 ## Configuration in Spring Boot
 
-Axon will automatically provide certain Kafka related components based on the availability of beans and/or properties.
+This extension can be added as a Spring Boot starter dependency to your project using group id
+ `org.axonframework.extensions.kafka` and artifact id `axon-kafka-spring-boot-starter`.
+When using the auto configuration, the following components will be created for you automatically:
+
+**Generic Components:**
+ * A `DefaultKafkaMessageConverter` using the configured `eventSerializer` (which defaults to `XStreamSerializer`).
+ Uses a `String` for the keys and a `byte[]` for the record's values
+ 
+**Producer Components:**
+ * A `DefaultProducerFactory` using a `String` for the keys and a `byte[]` for the record's values.
+ This creates a `ProducerFactory` in confirmation mode "NONE", as is specified [here](#publishing-events-to-kafka).
+ The `axon.kafka.publisher.confirmation-mode` should be adjusted to change this mode,
+  where the "TRANSACTIONAL" mode requires `axon.kafka.producer.transaction-id-prefix` property should be provided.
```

Comment:
`requires ... property should be provided.` -> `requires ... property to be provided.`